src/main/java/org/prebid/server/bidder/bidmachine/BidmachineBidder.java

```diff
@@ -0,0 +1,235 @@
+            if (statusCode == 204) {
+                return Result.empty();
+            }
+
+            if (statusCode != 200) {
+                return Result.withError(BidderError.badServerResponse(
+                        "unexpected status code: " + statusCode + " " + httpCall.getResponse().getBody()
+                ));
+            }
+
+            final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+            return Result.of(extractBids(httpCall.getRequest().getPayload(), bidResponse, errors), errors);
+        } catch (DecodeException | PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
+    }
+
+    private List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse, List<BidderError> errors) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
+        }
+        return bidsFromResponse(bidRequest, bidResponse, errors);
+    }
+
+    private static List<BidderBid> bidsFromResponse(BidRequest bidRequest, BidResponse bidResponse,
+                                                    List<BidderError> errors) {
+        return bidResponse.getSeatbid().stream()
+                .filter(Objects::nonNull)
+                .map(SeatBid::getBid)
+                .filter(Objects::nonNull)
+                .flatMap(Collection::stream)
+                .map(bid -> BidderBid.of(bid,
+                        getBidType(bid.getImpid(), bidRequest.getImp(), errors), bidResponse.getCur()))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
+    }
+
+    private static BidType getBidType(String impId, List<Imp> imps, List<BidderError> errors) {
+        for (Imp imp : imps) {
+            if (imp.getId().equals(impId)) {
+                if (imp.getBanner() == null && imp.getVideo() != null) {
+                    return BidType.video;
+                } else {
+                    return BidType.banner;
+                }
+            }
+        }
+
+        errors.add(BidderError.badServerResponse(String.format("Failed to find impression with id: %s", impId)));
+        return null;
+    }
+
+    private BidRequest createSingleRequest(Imp imp, ExtImpPrebid extPrebid, BidRequest request) {
+        return request.toBuilder()
+                .imp(Collections.singletonList(modifyImp(imp, extPrebid)))
+                .build();
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpPrebid extPrebid) {
+        if (extPrebid != null && extPrebid.getIsRewardedInventory() == 1) {
+            final Banner banner = imp.getBanner();
+            final List<Integer> bannerBattr = banner == null ? null : banner.getBattr();
+            final Imp.ImpBuilder impBuilder = imp.toBuilder();
+            if (bannerBattr != null && !hasRewardedBattr(bannerBattr)) {
```

Comment:
In Go they always add 16 even if bannerBattr or VideoBattr is not present, check `ListUtils.defaultIfNull` or same method for banner and videoBattr resolving